tensorpipe/core/message.h

```diff
@@ -49,12 +51,14 @@ class Message final {
   std::vector<Payload> payloads;
 
   struct Tensor {
-    void* data{nullptr};
-    size_t length{0};
-
-    // Users may include arbitrary metadata in the following fields.
+    tensorpipe::Tensor tensor;
```

Comment:
We should perhaps try to find a different name for this, or somehow reorganize this struct, as now users would do something like `msg.tensors[0].tensor` which is redundant...

Reply:
Agreed, I've just not yet found a convincing name. `data` would fit nicely, but as you know, it is currently being used already.

Reply:
As discussed, renaming `tensor` to `buffer`, and `CpuTensor` to `CpuBuffer`.